src/printer.js

```diff
@@ -3609,7 +3607,7 @@ function printMemberChain(path, options, print) {
       // beginning of the next one
       if (
         printedNodes[i].node.computed &&
-        isLiteral(printedNodes[i].node.property)
+        isNumericLiteral(printedNodes[i].node.property)
```

Comment:
I had to change this to numeric literals only, as

```js
method()
  .then(x => x)
  ["abc"](x => x)
  [abc](x => x);
```

was being formatted as

```
method()
  .then(x => x)["abc"](x => x)
  [abc](x => x);
```

Reply:
Was doing some digging and it turns out way back in #469 this was changed to be a single line, there's even discussion here: https://github.com/prettier/prettier/pull/469/files#r97891989.

@vjeux You mentioned:
> there's likely going to be a function with `{}` which will trigger everything to expand.

Do you have the issue number for why this is no longer the case?